engine/src/main/java/org/destinationsol/FactionDisplay.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Input;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.math.Vector2;
+import org.destinationsol.game.*;
+import org.destinationsol.game.screens.MapScreen;
+import org.destinationsol.game.ship.SolShip;
+import org.destinationsol.ui.SolInputManager;
+import org.destinationsol.ui.UiDrawer;
+
+public class FactionDisplay {
+    private SolCam cam;
+    private FactionInfo info;
+    private boolean isPressed = false;
+    private MapScreen mapScreen;
```

Comment:
For whatever reason it's giving me an error whenever I go for MapScreen directly. Or even when defining it in the draw faction names method.

Reply:
But you never assign anything to that field! And neither you ever use it!

Reply:
Yeah I'm really not sure why it won't compile if I do otherwise. isScreenOn(MapScreen) gives me an error, and so does defining MapScreen mapScreen and using isScreenOn(mapScreen).

Reply:
Is this resolved? It looks like it remains an issue.

The reason line `if (isPressed && cam.getAngle() == 0 && !inputManager.isScreenOn(mapScreen)) {` works is like because `mapScreen` does exist and gets initialized by default to `null`. 

Then in `inputManager.isScreenOn` we check and see if there's a match in currently contained screens for `null` which naturally ends up returning `false` (well, I assume), meaning the logic in `drawFactionNames` runs.

The goggles, they do nothing! Same as that condition :-)